firedrake/paraview_reordering.py

```diff
@@ -0,0 +1,449 @@
+from tsfc.fiatinterface import create_element
+import numpy as np
+
+
+def firedrake_local_to_cart(element):
+    r"""Gets the list of nodes for an element (provided they exist.)
+    :arg element: a ufl element.
+    :returns: a list of arrays of floats where each array is a node.
+    """
+    fiat_element = create_element(element, vector_is_mixed=False)
+    # TODO: Surely there is an easier way that I've forgotten?
+    carts = [np.array(list(phi.get_point_dict().keys())[0])
+             for phi in fiat_element.dual_basis()]
+    return carts
+
+
+def invert(list1, list2):
+    r"""Given two maps (lists) from [0..N] to nodes, finds a permutations between them.
+    :arg list1: a list of nodes.
+    :arg list2: a second list of nodes.
+    :returns a list of integers, l, such that list1[x] = list2[l[x]]
+    """
+    if len(list1) != len(list2):
+        raise Exception("Dimension of Paraview basis and Element basis unequal.")
+
+    def find_same(val, lst, tol=0.0000001):
+        for (idx, x)in enumerate(lst):
+            if np.linalg.norm(val - x) < tol:
+                return idx
+        raise Exception("Basis of element incompatible with paraview basis: node {0} is not in {1}".format(val, lst))
+    # finds map from idx of list 1 to list of list 2.
+    perm = [find_same(x, list2) for x in list1]
+    #  check for bijectivity:
+    if len(set(perm)) != len(list1):
+        raise Exception("Unable to establish permutation between Paraview basis and given element basis.")
+    return perm
+
+
+"""
+The following functions are translations of funtions in the VTK source;we use them find the order of nodes in the lagrange basises that Paraview uses. We don't document them fully, but link back to the vtk source.
+These come from VTK version 8.2.0.
+We could simplify all of this by depending on VTK's python bindings.
```

Comment:
`pip install vtk` appears to work, the only question for me is whether it plays badly with any global vtk install

Reply:
I don't think it will. 1. Anecdotally, I have no issues with a global install and python bindings. 2. The python bindings come with their own binary blobs (dynamic libs) that are, unless you got out of your way, only used by the python bindings (just look at https://pypi.org/project/vtk/#files).